src/sentry/integrations/github/client.py

```diff
@@ -188,6 +186,11 @@ def get_trees_for_org(
                 repo_files = self.get_cached_repo_files(full_name, branch)
                 trees[full_name] = RepoTree(Repo(full_name, branch), repo_files)
             logger.info("Using cached trees for Github org.", extra=extra)
+        except ApiError as e:
```

Comment:
If any of the calls to `get_cached_repo_files` produces an API rate limit handle it in this code block and do not try anymore.

Ideally we should block other tasks from trying before the API rate limit is reset but this is already an improvement and may prevent Github blocking the installation.